python-sdk/src/astro/databases/mssql.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING
+
+import pandas as pd
+import sqlalchemy
+from airflow.providers.microsoft.mssql.hooks.mssql import MsSqlHook
+from sqlalchemy.sql import ClauseElement
+
+from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy, MergeConflictStrategy
+from astro.databases.base import BaseDatabase
+from astro.settings import MSSQL_SCHEMA
+from astro.table import BaseTable, Metadata
+
+DEFAULT_CONN_ID = MsSqlHook.default_conn_name
+if TYPE_CHECKING:  # pragma: no cover
+    from sqlalchemy.engine.cursor import CursorResult
+
+
+class MssqlDatabase(BaseDatabase):
+    DEFAULT_SCHEMA = MSSQL_SCHEMA
+
+    _create_schema_statement: str = """
+    IF NOT EXISTS (SELECT 1 FROM sys.schemas WHERE name = '{0}')
+    BEGIN
+        EXEC( 'CREATE SCHEMA {0}' );
+    END
+    """
+    _truncate_table_statement: str = "TRUNCATE TABLE {0}"
+
+    # CTAS IS NOT SUPPORTED IN SQL SERVER. HENCE USE SELECT * INTO
+    # SELECT * INTO FROM A SUBQUERY WILL WORK IN SQL SERVER ONLY WHEN AN ALIAS IS ADDED
+    _create_table_statement: str = """
+    IF (NOT EXISTS (SELECT *
+                 FROM INFORMATION_SCHEMA.TABLES
+                 WHERE TABLE_SCHEMA = '{0}'
+                 AND   TABLE_NAME    = '{1}'
+                 AND   TABLE_CATALOG = '{2}'))
+    BEGIN
+        SELECT * INTO  {2}.{0}.{1} FROM ({3}) as astro;
+    END
+    """
+
+    _queries_requiring_autocommit: list = ["alter", "select * into"]
+
+    def __init__(self, conn_id: str = DEFAULT_CONN_ID, table: BaseTable | None = None):
+        super().__init__(conn_id)
+        self.table = table
+
+    @property
+    def hook(self) -> MsSqlHook:
+        """Retrieve Airflow hook to interface with the mssql database."""
+        kwargs = {}
+        _hook = MsSqlHook(mssql_conn_id=self.conn_id)
+        if (_hook.schema is None) and (self.table and self.table.metadata and self.table.metadata.database):
+            kwargs.update({"schema": self.table.metadata.database})
+        return MsSqlHook(mssql_conn_id=self.conn_id, **kwargs)
+
+    @property
+    def sql_type(self) -> str:
+        return "mssql"
+
+    def table_exists(self, table: BaseTable) -> bool:
+        """
+        Check if a table exists in the database
+        :param table: Details of the table we want to check that exists
+        """
+        inspector = sqlalchemy.inspect(self.sqlalchemy_engine)
+
+        schema = None
+        if table and table.metadata:
+            schema = table.metadata.schema
+        # elif self.hook.schema:
+        #     schema = self.hook.schema
+        return bool(inspector.dialect.has_table(self.connection, table.name, schema=schema))
```

Comment:
a fully qualified name is not working here? 
like: return bool(inspector.dialect.has_table(self.connection, table_qualified_name))

Reply:
It doesnt work because the database would contain the name as `abc` instead of `db.schema.abc`.